lib/fizzbuzz_advanced.rb

```diff
@@ -0,0 +1,24 @@
+class FizzbuzzAdvanced
+  def initialize
+    @values = {
+      3 => "Fizz",
+      5 => "Buzz",
+      7 => "Sivv",
+      13 => "Grrr"
+    }
+  end
+
+  def calculate(num)
+    to_return = ""
+    output = ""
+
+    @values.each do |k, v|
+      output += v if num % k == 0
```

Comment:
I get a little tripped up on reading one-letter iterator variables when there are two of them. Maybe `factor` and `word`?